mod-source-record-storage-server/src/main/java/org/folio/consumers/AuthorityDomainKafkaHandler.java

```diff
@@ -0,0 +1,103 @@
+package org.folio.consumers;
+
+import static org.folio.dao.util.RecordDaoUtil.filterRecordByExternalId;
+
+import io.vertx.core.Future;
+import io.vertx.core.json.JsonObject;
+import io.vertx.kafka.client.consumer.KafkaConsumerRecord;
+import java.util.Collections;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.dao.util.RecordType;
+import org.folio.kafka.AsyncRecordHandler;
+import org.folio.rest.jooq.enums.RecordState;
+import org.folio.services.RecordService;
+import org.folio.services.util.KafkaUtil;
+import org.springframework.stereotype.Component;
+
+@Component
+public class AuthorityDomainKafkaHandler implements AsyncRecordHandler<String, String> {
+
+  private static final Logger log = LogManager.getLogger();
+
+  private static final String DOMAIN_EVENT_TYPE_HEADER = "domain-event-type";
+  private static final String DELETE_DOMAIN_EVENT_TYPE = "DELETE";
+  private static final String DELETE_EVENT_SUB_TYPE_FLD = "deleteEventSubType";
+  private static final String TENANT_FLD = "tenant";
+
+  private final RecordService recordService;
+
+  public AuthorityDomainKafkaHandler(RecordService recordService) {
+    this.recordService = recordService;
+  }
+
+  @Override
+  public Future<String> handle(KafkaConsumerRecord<String, String> consumerRecord) {
+    log.trace("handle:: Handling kafka record: '{}'", consumerRecord);
+    String authorityId = consumerRecord.key();
+    if (isUnexpectedDomainEvent(consumerRecord)) {
+      log.trace("handle:: Expected only {} domain type. Skipping authority domain kafka record [ID: '{}']",
```

Comment:
it the domain event is unexpected, should a `return` be here?

Reply:
Yes, fixed.